service/src/test/java/skills/intTests/metrics/skill/AchievementsByTagMetricsBuilderSpec.groovy

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2021 SkillTree
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package skills.intTests.metrics.skill
+
+import org.springframework.beans.factory.annotation.Autowired
+import skills.intTests.utils.DefaultIntSpec
+import skills.intTests.utils.SkillsFactory
+import skills.intTests.utils.SkillsService
+import skills.metrics.builders.skill.AchievementsByTagMetricsBuilder
+
+class AchievementsByTagMetricsBuilderSpec  extends DefaultIntSpec {
+    @Autowired
+    AchievementsByTagMetricsBuilder builder
+
+    def "produces accurate achievement user list"() {
```

Comment:
enhance data set to create multiple skills; users must have different achievements under multiple skills; validate that different skills return proper chart/metric data